api/v3/utils.php

```diff
@@ -1538,6 +1538,89 @@ function _civicrm_api3_custom_data_get(&$returnArray, $checkPermission, $entity,
 }
 
 /**
+ * Used by the Validate API.
+ * @param string $entity
+ * @param string $action
+ * @param array $params
+ *
+ * @return array $errors
+ */
+function _civicrm_api3_validate($entity, $action, $params) {
+  $errors = array();
+  $fields = civicrm_api($entity, 'getfields', array('version' => 3, 'action' => $action));
+
+  // Check for required fields here.
+
+  // Select only the fields which have been input as a param.
+  $fields = array_intersect_key($fields, $params);
+
+  // This derives heavily from the function "_civicrm_api3_validate_fields".
+  // However, the difference is that try-catch blocks are nested in the loop, making it
+  // possible for us to get all errors in one go.
+  foreach ($fields as $fieldName => $fieldInfo) {
+    switch (CRM_Utils_Array::value('type', $fieldInfo)) {
+      case CRM_Utils_Type::T_INT:
+        try {
+          _civicrm_api3_validate_integer($params, $fieldName, $fieldInfo, $entity);
+        }
+        catch (Exception $e) {
+          $errors[] = $e->getMessage();
```

Comment:
I think you want to key it by field.

Also should we go for an array - ie 

 $errors['financial_type_id'] = array('code' => 'Invalid Integer', 'message' => $e->getMessaget());

?